### ISSUE TITLE
Unprivileged Users Not Supported?

### ISSUE THREAD

**Issue by cyberfreaq** [hausec/PowerZure#11]
I think since the merge of the dev branch it is no longer possible to use PowerZure with users who do not have an active subscription. During initialization PowerZure throws the error `Get-AzRoleAssignment : Object reference not set to an instance of an object.`

I think, most of the users "owned" during an assessment won't have any active subscriptions. Wouldn't it be better, if you could also query Azure/Azure AD with those users (or at least the parts of Azure, to which the users possibly have some access to) or am I missing something here?

**Comment by halfluke** [comment]
there is something strange which at this point I am not sure is related to this issue you have raised or to the one you created a fix for.
When I tried your patched version for the first time I got an error about "an obect cannot be empty", and I noticed it happened because SubscriptionName TenantId fields were empty (after using connect-azaccount)
I am not sure why those fields are not populated sometimes, I've deleted the 3 json files  AzureRmContext AzurePSDataCollectionProfile AzureRmContextSettings, run connect-azaccount, fields are populated and all is good.
And I cannot reproduce the error.
But yes, it looks like that if you do not have a subscription (or powershell cannot retrieve it) you run into issues (basically you cannot use PowerZure)
